components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -966,31 +971,24 @@ fun stringOrNull(jsonObject: JSONObject, key: String): String? {
 }
 
 data class SearchResult(
-    val searchString: String,
     val url: String,
     val title: String,
-    val frecency: Long,
-    val iconUrl: String? = null
+    val frecency: Long
     // Skipping `reasons` for now...
 ) {
     companion object {
-        fun fromJSON(jsonObject: JSONObject): SearchResult {
-            return SearchResult(
-                searchString = jsonObject.getString("search_string"),
-                url = jsonObject.getString("url"),
-                title = jsonObject.getString("title"),
-                frecency = jsonObject.getLong("frecency"),
-                iconUrl = stringOrNull(jsonObject, "icon_url")
-            )
-        }
-
-        fun fromJSONArray(jsonArrayText: String): List<SearchResult> {
-            val result: MutableList<SearchResult> = mutableListOf()
-            val array = JSONArray(jsonArrayText)
-            for (index in 0 until array.length()) {
-                result.add(fromJSON(array.getJSONObject(index)))
+        fun fromMessage(msg: MsgTypes.SearchResultList): List<SearchResult> {
+            val results: MutableList<SearchResult> = mutableListOf()
+            for (index in 0 until msg.getResultsCount()) {
```

Comment:
I think there's an array that you can get, `.results` or `.result`, here's an example of how we do collections in general:
https://github.com/mozilla/application-services/blob/2649158b61c4e6d7f87bd2e3f16553393094674b/components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/Device.kt#L77-L96

Reply:
Done